frontend/javascripts/dashboard/dataset/dataset_cache_provider.js

```diff
@@ -155,9 +168,41 @@ export default function DatasetCacheProvider({ children }: { children: Node }) {
     }
   }
 
+  async function updateCachedDataset(dataset: APIDataset) {
+    if (isLoading) return;
+    setIsLoading(true);
```

Comment:
Thanks for spotting this. I implemented a queueing like behaviour for all updates on the same dataset. This queuing ensures that each previous request is finished before the current request starts. Therefore it is ensured that the backend always gets the updates in the correct order. Thus no update call that (by network magic) was faster at the server gets lost because a slower update request that is much slower overwrites its changes afterwards.
Here is a screenshot of a few dataset changes made on a single dataset in a row by changing the tags of a single dataset. The connection is throttled very strongly to test this corner case. The graph on the right illustrates that one call only starts after the previous updates finished.
![image](https://user-images.githubusercontent.com/39529669/144451685-30c4a214-55ac-49ef-8f92-d03b994d331a.png)